specs/AppLifecycle/Instancing/AppLifecycle SingleMulti-Instancing.md

```diff
@@ -0,0 +1,422 @@
+information about each other to make a decision about single-instancing, they often use named pipes.
+
+Conversely, UWP apps are single-instanced by default. Windows 10 1803 introduced support for UWP
+apps to opt in to be multi-instanced. This also allowed for the possibility that even though an app
+has chosen to be multi-instanced, there might be some scenario where it wants to redirect a fresh
+activation request to an existing instance instead.
+
+## Differences between Windows and Reunion Instancing
+
+The Reunion behavior is based on the existing
+[Windows.ApplicationModel.AppInstance](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.appinstance)
+class, but with some key differences:
+
+### List of Instances
+
+-   **Platform**: GetInstances returns only the instances that the app explicitly registered for
+    potential redirection.
+-   **Reunion**: GetInstances returns all running instances of the app.
+
+### Registering Keys
+
+Each instance of the app can register a key, and this key is based on arbitrary app-defined logic.
+The app is allowed to set its key as many times as it likes; and each time we will overwrite the
+data for that instance in the cache. A given key can only be registered for one instance. If an app
+attempts to register a key that has already been registered for a different instance, the API will
+return the registered instance, and the app can check whether it is the current instance.
+
+-   **Platform**: An instance must register a key in order to be included in the list returned from
+    GetInstances.
+-   **Reunion**: Registering a key is decoupled from the list of instances.
+
+### Instance Redirection Targets
+
+An instance of an app can take part in "activation redirection". This means that this instance of
+the process can handle multiple Activated events, and so it could be a target for selective
+activation redirection. When an activation is redirected to another instance, it arrives at that
+instance's Activated callback, thus hooking into the normal activation workflow that apps already
+expect. An app can have multiple instances available as redirection targets.
+
+-   **Platform**: Only instances that have registered a key can be a target for redirection.
+-   **Reunion**: Any instance can be a redirection target, whether or not it has a registered key.
+
+### Post-Redirection Behavior
+
+Redirection is a terminal operation. After calling the method to redirect, there is nothing the app
+can do - regardless of whether the redirect succeeded or failed. In calling redirect, the app is
+saying that it cannot/will not handle this activation request - so even if the redirection fails,
+there's nothing this instance can or will do anyway. The platform will explicitly terminate an
+instance that has chosen to redirect. If redirection fails, the activation request fails. The UX is
+the same as an activation failure today.
```

Comment:
Can the redirected-to app perform another redirect to pass the request further? Or is it required to handle it directly?

E.g., "The previous instance redirected to me because I had an existing Contoso document open, but due to a race condition, by the time I received the activation, I'm no longer viewing a Contoso document, but I know someone else who is, so I want to re-redirect to them."

Reply:
Expanded description.